apisix/event.lua

```diff
@@ -0,0 +1,70 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+
+local require = require
+local error   = error
+local ngx     = ngx
+
+local _M = {}
+
+_M.EVENTS_MODULE_LUA_RESTY_WORKER_EVENTS = 'lua-resty-worker-events'
+_M.EVENTS_MODULE_LUA_RESTY_EVENTS = 'lua-resty-events'
+
+
+-- use lua-resty-worker-events
+local function init_worker_events()
+    local we = require("resty.worker.events")
+    local shm = ngx.config.subsystem == "http" and "worker-events" or "worker-events-stream"
+    local ok, err = we.configure({shm = shm, interval = 0.1})
+    if not ok then
+        error("failed to init worker event: " .. err)
+    end
+
+    return we
+end
+
+
+function _M.init_worker()
```

Comment:
should we pass the option?

Reply:
@monkeyDluffy6017 Do you mean some options for the events module, such as the configuration from yaml?

Reply:
Yeah,  it's better if we can do that in the `config.yaml`

Reply:
It looks like we don't need any configuration for existing use cases. Perhaps we'll need to configure that later when we provide a switchable events module implementation.

Reply:
@monkeyDluffy6017 To summarize, we may introduce some options in future PR, if you have concerns about the continuity of implementation, we can create a separate new branch for staging each PR and commit the integrated code to the mainline from the new branch.